javascript/ql/test/query-tests/Security/CWE-022/TaintedPath/normalizedPaths.js

```diff
@@ -249,3 +249,26 @@ app.get('/resolve-path', (req, res) => {
   else
     fs.readFileSync(path); // NOT OK - wrong polarity
 });
+
+app.get('/relative-startswith', (req, res) => {
+  let path = pathModule.resolve(req.query.path);
+
+  fs.readFileSync(path); // NOT OK
+
+  var self = something();
+	
+  var relative = pathModule.relative(self.webroot, path);
+  if(relative.startsWith(".." + pathModule.sep) || relative == "..") {
+    fs.readFileSync(path); // NOT OK! 
+  } else {
+    fs.readFileSync(path); // OK! 
+  }
+
+  let newpath = pathModule.normalize(p);
+  var relativePath = path.relative(path.normalize(workspaceDir), newpath);
```

Comment:
In QL, do we want to support the pattern of `path.normalize(path.relative(x, newpath));`? Programmers may want to be *extra* safe that they are doing the right thing, so the redundancy is not completely unrealistic IMO.

Reply:
I didn't find anyone using that pattern, but it doesn't sound unreasonable, so I added support for it.